chal_types/ai_generation.py

```diff
@@ -0,0 +1,50 @@
+import openai
+import random
+
+def gen_html(prompt):
+    openai.api_key = '<YOUR-API-KEY-HERE>'
+    
+    response = openai.Completion.create(
+        engine='text-davinci-003',
+        prompt=prompt,
+        max_tokens=2000,
+        temperature=0.7,
+        top_p=1.0,
+        frequency_penalty=0.0,
+        presence_penalty=0.0,
+        stop=None,
+        n=1
+    )
+
+    return response.choices[0].text.strip()
+
+def format_links(file):
+    formatted_links = []
+    x = open(file, "r")
+
+    links = x.readlines()
+    i = 0
+    for l in links:
+        if(l != "\n"):
+            formatted_links.append(l.strip("\n"))
+            i+=1
+    return formatted_links
+
+def add_imgs(html, links):
+    while True:
+        if "[PLACEHOLDER]" in html:
```

Comment:
this is fine for now, but i think we want to integrate a template engine like jinja

Reply:
I could do that, but I feel it would be limiting. At least to my understanding, to replace jinja tags you have to specify them in a function, but with chatgpt making different sites every time, the amount of tags will differ. The only solution I can see is asking chatgpt to generate a set number of tags, unless there's some way to dynamically replace them that I don't know about.

Reply:
yeah that is the whole point of jinja, i would recommend reading this tutorial https://realpython.com/primer-on-jinja-templating/